### PR TITLE
Curator

### PR DESCRIPTION
**Pull request by JasonTheMain**
## Changes proposed in this pull request:

- Addition of Curator package
- Addition of Curator job
- Addition of Blobs for Curator

## Things to check

- For any logging statements, is there any chance that they could be logging sensitive data?
- Are log statements using a logging library with a logging level set? Setting a logging level means that log statements "below" that level will not be written to the output. For example, if the logging level is set to `INFO` and debugging statements are written with `log.debug` or similar, then they won't be written to the otput, which can prevent unintentional leaks of sensitive data.

## Security considerations

none

### CODE REVIEW

File: jobs/curator/templates/config/admin-crt.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<% if_link('opensearch') do |curator| %>
+<%= curator.p('opensearch.admin.certificate', '') %>
```
Comment (markdboyd) [inline jobs/curator/templates/config/admin-crt.erb:2]:
```suggestion
<% if_link('opensearch') do |opensearchconfig| %>
<%= opensearchconfig.p('opensearch.admin.certificate', '') %>
```

File: jobs/curator/templates/config/admin-pem.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<% if_link('opensearch') do |curator| %>
+<%= curator.p('opensearch.admin.private_key', '') %>
```
Comment (markdboyd) [inline jobs/curator/templates/config/admin-pem.erb:2]:
```suggestion
<% if_link('opensearch') do |opensearchconfig| %>
<%= opensearchconfig.p('opensearch.admin.private_key', '') %>
```

File: jobs/curator/templates/config/ca.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<% if_link('opensearch') do |curator| %>
+<%= curator.p('opensearch.node.ssl.ca', '') %>
```
Comment (markdboyd) [inline jobs/curator/templates/config/ca.erb:2]:
```suggestion
<% if_link('opensearch') do |opensearchconfig| %>
<%= opensearchconfig.p('opensearch.node.ssl.ca', '') %>
```

File: packages/curator/packaging [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+set -e
+
+export PATH="/var/vcap/packages/python3/bin:${PATH}" LD_LIBRARY_PATH="/var/vcap/packages/python3/lib:${LD_LIBRARY_PATH}" C_INCLUDE_PATH="/var/vcap/packages/python3/include:${C_INCLUDE_PATH}"
+
+# --no-index prevents contacting pypi to download packages
+# --find-links tells pip where to look for the dependancies
+# --prefix installation prefix where lib, bin and other top-level folders are placed
+pip3 install --no-index "--prefix=${BOSH_INSTALL_TARGET}" curator/vendor/flit_core-3.8.0.tar.gz
+pip3 install --no-index --find-links ./curator/vendor/ "--prefix=${BOSH_INSTALL_TARGET}" curator/vendor/wheel-0.41.2.tar.gz
+# pip3 install --no-index "--prefix=${BOSH_INSTALL_TARGET}" curator/vendor/setuptools-67.8.0.tar.gz
```
Comment (markdboyd) [inline packages/curator/packaging:11]:
```suggestion
```

File: jobs/curator/templates/config/bpm.yml.erb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+processes:
+  - name: curator
+    hooks:
+      pre_start: /var/vcap/jobs/curator/bin/pre-start
+    ephemeral_disk: true
+    additional_volumes:
+    - path: /var/vcap/jobs/opensearch/config
+      writable: true
```
Comment (markdboyd) [inline jobs/curator/templates/config/bpm.yml.erb:8]: Why is this here? Is this job reading from the opensearch job config? If not we should remove this

Comment (JasonTheMain) [reply jobs/curator/templates/config/bpm.yml.erb:8]: We needed to pass it the certs, i can try it without

Comment (markdboyd) [reply jobs/curator/templates/config/bpm.yml.erb:8]: It looks like the certs come from the `curator` job now?

https://github.com/cloud-gov/opensearch-boshrelease/blob/7f883cbc0e9ad85ec82e957f02c3cab7c144e8f3/jobs/curator/templates/config/config.yml.erb#L7-L9

File: jobs/curator/spec [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+packages:
+  - curator
+  - python3
+
+templates:
+  bin/pre-start: bin/pre-start
+  bin/run-curator: bin/run-curator
+  config/actions.yml.erb: config/actions.yml
+  config/config.yml.erb: config/config.yml
+  config/bpm.yml.erb: config/bpm.yml
```
Comment (markdboyd) [inline jobs/curator/spec:15]: Needs to be removed

**Review by markdboyd — APPROVED** [review]
